docarray/array/array/io.py

```diff
@@ -530,3 +598,85 @@ def save_binary(
             file_ctx=file_ctx,
             show_progress=show_progress,
         )
+
+
+def is_access_path_valid(doc: Type['BaseDocument'], access_path: str) -> bool:
+    """
+    Check if a given access path is a valid path for a given Document class.
+    """
+    field, _, remaining = access_path.partition('__')
```

Comment:
Didn't we use "dot-notation" in other places, e.g. `nested.field` instead of `nested__field`? We need to make sure that this is uniform everywhere.